lib/storage/src/chunks/getChunkStream.ts

```diff
@@ -0,0 +1,39 @@
+import { RawDataPart } from "../Upload";
+import { Buffer } from "buffer";
+
+interface Buffers {
+  chunks: Buffer[];
+  length: number;
+}
+
+export async function* getChunk<T>(
+  data: T,
+  partSize: number,
+  getNextData: (data: T) => AsyncGenerator<Buffer>
+): AsyncGenerator<RawDataPart> {
+  let partNumber = 1;
+  const currentBuffer: Buffers = { chunks: [], length: 0 };
+
+  for await (const datum of getNextData(data)) {
+    currentBuffer.chunks.push(datum);
+    currentBuffer.length += datum.length;
+
+    while (currentBuffer.length >= partSize) {
+      // concat all the buffers together once.
+      const dataChunk: Buffer = Buffer.concat(currentBuffer.chunks);
```

Comment:
it look like to me that it should be done before the while loop. Indeed current.chunks is a single element array after the first iteration (line 30) so Buffer.concat seems useless for the 2nd and next iterations

Reply:
You're right, nice catch, lemme wrap this in a branch

Reply:
done